syntax = "proto3";

package bookstore;

import "shared/order.proto";
import "suggestions/suggestions.proto";

service TransactionVerificationService {
    rpc InitVerifyTransaction (TransactionRequest) returns (ErrorResponse);
    rpc VerifyTransaction (ExecInfo) returns (ErrorResponse);
    rpc DeleteOrder (ExecInfo) returns (ErrorResponse);
}

message TransactionRequest {
    ExecInfo info = 1;
    string name = 2;
    CreditCard CreditCard = 3;
    BillingAddress BillingAddress = 4;
    repeated Book booksInCart = 5;
}

message TransactionResponse {
    bool transactionVerified = 1;
}
